src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerManager.AnalyzerExecutionContext.cs

```diff
@@ -17,6 +18,11 @@ internal partial class AnalyzerManager
     {
         private sealed class AnalyzerExecutionContext
         {
+            /// <summary>
+            /// Cached mapping of localizable strings in this descriptor to any exceptions thrown while obtaining them.
+            /// </summary>
+            private static ImmutableSegmentedDictionary<LocalizableString, Exception?> s_localizableStringToException = ImmutableSegmentedDictionary<LocalizableString, Exception?>.Empty.WithComparer(Roslyn.Utilities.ReferenceEqualityComparer.Instance);
```

Comment:
There's not a risk of this growing unbounded or anything like that, right?

Reply:
it's effectively capped at "num analyzer * 3".  if we have an unbounded number of analyzers, we're def screwed anyways :)